assignment/models/homework.py

```diff
@@ -0,0 +1,244 @@
+                                  help='File format')
+    attach_files = fields.Char("File Name", help='Enter file name')
+    subject_standard_assignment = fields.Many2one("standard.standard",
+                                                  help='''Standard of the 
+                                                  assignment in which it 
+                                                  assigned''')
+
+    @api.onchange('standard_id')
+    def onchange_subject_standard(self):
+        '''Onchange method to assign assignment based on standard'''
+        self.subject_standard_assignment = self.standard_id.standard_id.id
+
+    @api.constrains('assign_date', 'due_date')
+    def check_date(self):
+        '''Method to check constraint of due date and assign date'''
+        if self.due_date < self.assign_date:
+            raise ValidationError(_('''
+                Due date of homework should be greater than assign date'''))
+
+    def active_assignment(self):
+        ''' This method change state as active state
+            and create assignment line
+            @return : True
+        '''
+        assignment_obj = self.env['school.student.assignment']
+        student_obj = self.env['student.student']
+        ir_attachment_obj = self.env['ir.attachment']
+        for rec in self:
+            student_recs = student_obj.search([
+                            ('standard_id', '=', rec.standard_id.id),
+                            ('state', '=', 'done')])
+            if not rec.attached_homework:
+                raise ValidationError(_('''Please attach the homework!'''))
+            for std in student_recs:
+                ass_dict = {'name': rec.name,
+                            'subject_id': rec.subject_id.id,
+                            'standard_id': rec.standard_id.id,
+                            'assign_date': rec.assign_date,
+                            'due_date': rec.due_date,
+                            'state': 'active',
+                            'attached_homework': rec.attached_homework,
+                            'teacher_id': rec.teacher_id.id,
+                            'teacher_assignment_id': rec.id,
+                            'student_id': std.id,
+                            'stud_roll_no': std.roll_no,
+                            'student_standard': std.standard_id.standard_id.id,
+                            'submission_type': self.type_submission,
+                            'attachfile_format': self.file_format.name}
+                assignment_rec = assignment_obj.create(ass_dict)
+                attach = {'name': 'test',
+                          'datas': rec.attached_homework,
+                          'description': 'Assignment attachment',
+                          'res_model': 'school.student.assignment',
+                          'res_id': assignment_rec.id}
+                ir_attachment_obj.create(attach)
+            rec.write({'state': 'active'})
```

Comment:
If the method is not callable from a different function, rec.state='active' will work